datadog_checks_dev/datadog_checks/dev/tooling/e2e/local.py

```diff
@@ -0,0 +1,136 @@
+        elif ON_WINDOWS:
+            return WINDOWS
+        else:
+            raise Exception("Unsupported OS for Local E2E")
+
+    @property
+    def agent_version(self):
+        return self._agent_version or DEFAULT_AGENT_VERSION
+
+    @property
+    def agent_command(self):
+        return '{}'.format(get_agent_exe(self.agent_version, platform=self.platform))
+
+    def write_config(self):
+        write_env_data(self.check, self.env, self.config, self.metadata)
+        self.copy_config_to_local_agent()
+
+    def remove_config(self):
+        remove_env_data(self.check, self.env)
+        self.remove_config_from_local_agent()
+
+    def copy_config_to_local_agent(self):
+        conf_dir = get_agent_conf_dir(self.check, self.agent_version, self.platform)
+        check_conf_file = os.path.join(get_agent_conf_dir(conf_dir, '{}.yaml'.format(self.check)))
+        if not os.path.exists(conf_dir):
+            os.makedirs(conf_dir)
+
+        if file_exists(check_conf_file):
+            copyfile(check_conf_file, '{}.bak'.format(check_conf_file))
+        copyfile(self.config_file, check_conf_file)
+
+    def remove_config_from_local_agent(self):
+        check_conf_file = os.path.join(
+            get_agent_conf_dir(self.check, self.agent_version, self.platform),
+            '{}.yaml'.format(self.check)
+        )
+        backup_conf_file = '{}.bak'.format(check_conf_file)
+        os.remove(check_conf_file)
+        if file_exists(backup_conf_file):
+            move(backup_conf_file, check_conf_file)
+
+    def run_check(self, capture=False, rate=False):
+        command = '{} check {}{}'.format(
+            self.agent_command,
+            self.check,
+            ' {}'.format(get_rate_flag(self.agent_version)) if rate else ''
+        )
+        return run_command(command, capture=capture)
+
+    def update_check(self):
+        value = click.confirm('Env will started with an editable check install for the {} package. \
+This check will remain in an editable install after \
+the environment is torn down. Would you like to proceed? '.format(self.check))
+        if value:
+            install_cmd = AGENT_CMD[self.platform]['pip_install'] + ['-e',  path_join(get_root(), self.check)]
+            return run_command(install_cmd, capture=True, check=True)
+        else:
+            return
+
+    def update_base_package(self):
+        value = click.confirm('Env will started with an editable check install for the base package. \
+This check will remain in an editable install after \
+the environment is torn down. Would you like to proceed? ')
+        if value:
+            install_cmd = AGENT_CMD[self.platform]['pip_install'] + ['-e', self.base_package]
+            run_command(install_cmd, capture=True, check=True)
+        else:
+            return
+
+    def update_agent(self):
+        # The Local E2E assumes an Agent is already installed on the machine
+        echo_info("Using the locally installed Agent")
```

Comment:
This isn't the place for cli stuff, let's put this in the command or just `pass`